tests/nodeLabeller_test.go

```diff
@@ -0,0 +1,166 @@
+	"time"
+)
+
+var (
+	nlClusterRoleRes = &testResource{
+		Name:       nodelabeller.ClusterRoleName,
+		resource:   &rbac.ClusterRole{},
+		Namsespace: "",
+	}
+	nlClusterRoleBindingRes = &testResource{
+		Name:       nodelabeller.ClusterRoleBindingName,
+		resource:   &rbac.ClusterRoleBinding{},
+		Namsespace: "",
+	}
+	nlServiceAccountRes = &testResource{
+		Name:       nodelabeller.ServiceAccountName,
+		Namsespace: testNamespace,
+		resource:   &core.ServiceAccount{},
+	}
+	securityContextConstraintRes = &testResource{
+		Name:       nodelabeller.SecurityContextName,
+		Namsespace: "",
+		resource:   &secv1.SecurityContextConstraints{},
+	}
+	configMapRes = &testResource{
+		Name:       nodelabeller.ConfigMapName,
+		Namsespace: testNamespace,
+		resource:   &core.ConfigMap{},
+	}
+	daemonSetRes = &testResource{
+		Name:       nodelabeller.DaemonSetName,
+		Namsespace: testNamespace,
+		resource:   &apps.DaemonSet{},
+	}
+)
+
+var _ = Describe("Node Labeller", func() {
+	Context("resource creation", func() {
+		table.DescribeTable("created cluster resource", func(res *testResource) {
+			resource := res.NewResource()
+			err := apiClient.Get(ctx, client.ObjectKey{Name: res.Name}, resource)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(hasOwnerAnnotations(resource.GetAnnotations())).To(BeTrue())
+		},
+			table.Entry("[test_id:5193] cluster role", nlClusterRoleRes),
+			table.Entry("[test_id:5196] cluster role binding", nlClusterRoleBindingRes),
+			table.Entry("[test_id:5202] security context constraint", securityContextConstraintRes),
+		)
+
+		table.DescribeTable("created namespaced resource", func(res *testResource) {
+			err := apiClient.Get(ctx, client.ObjectKey{
+				Name: res.Name, Namespace: testNamespace,
+			}, res.NewResource())
+			Expect(err).ToNot(HaveOccurred())
+		},
+			table.Entry("[test_id:5205] service account", nlServiceAccountRes),
+			table.Entry("[test_id:5199] configMap", configMapRes),
+			table.Entry("[test_id:5190] daemonSet", daemonSetRes),
+		)
+	})
+
+	Context("resource deletion", func() {
+		table.DescribeTable("recreate after delete", func(res *testResource) {
+			resource := res.NewResource()
+			resource.SetName(res.Name)
+			resource.SetNamespace(res.Namsespace)
+			Expect(apiClient.Delete(ctx, resource)).ToNot(HaveOccurred())
+
+			Eventually(func() error {
+				return apiClient.Get(ctx, client.ObjectKey{
+					Name: res.Name, Namespace: res.Namsespace,
+				}, resource)
+			}, timeout, time.Second).ShouldNot(HaveOccurred())
+		},
+			table.Entry("[test_id:5194] cluster role", nlClusterRoleRes),
+			table.Entry("[test_id:5198] cluster role binding", nlClusterRoleBindingRes),
+			table.Entry("[test_id:5203] security context constraint", securityContextConstraintRes),
+			table.Entry("[test_id:5206] service account", nlServiceAccountRes),
+			table.Entry("[test_id:5200] configMap", configMapRes),
+			table.Entry("[test_id:5191] daemonSet", daemonSetRes),
+		)
+	})
+
+	Context("resource change", func() {
```

Comment:
[test_id:5207] service account reconciling after editing is missing

Reply:
The test for reconciliation of service account is probably not required. The reason being, the only two things that are set in the code for the sa are its name and namespace, and the k8s api server anyway does not let you change that. So there is no reconciliation test for it.